src/pages/cities/Forms/CityForm.js

```diff
@@ -7,173 +7,151 @@ import toast from 'react-hot-toast'
-              valueField='recordId'
-              displayField='name'
-              readOnly={editMode}
-              displayFieldWidth={1}
-              columnsInDropDown={[
-                { key: 'reference', value: 'Reference' },
-                { key: 'name', value: 'Name' }
-              ]}
-              values={formik.values}
-              required
-              maxAccess={maxAccess}
-              onChange={(event, newValue) => {
-                formik.setFieldValue('stateId', null)
-                if (newValue) {
-                  formik.setFieldValue('countryId', newValue?.recordId)
-                } else {
+          const res = await getRequest({
+            extension: SystemRepository.City.get,
+            parameters: `_recordId=${recordId}`
+          })
 
-                  formik.setFieldValue('countryId', '')
-                }
+          formik.setValues(res.record)
+        }
+      } catch {}
+      setIsLoading(false)
+    })()
+  }, [])
 
-              }}
-              error={formik.touched.countryId && Boolean(formik.errors.countryId)}
-              helperText={formik.touched.countryId && formik.errors.countryId}
-            />
+  return (
+    <FormShell resourceId={ResourceIds.Cities} form={formik} height={400} maxAccess={maxAccess} editMode={editMode}>
+      <Grid container spacing={4}>
+        <Grid item xs={12}>
+          <CustomTextField
+            name='reference'
+            label={labels.reference}
+            value={formik.values.reference}
+            required
+            maxAccess={maxAccess}
+            readOnly={editMode}
+            onChange={formik.handleChange}
+            onClear={() => formik.setFieldValue('reference', '')}
+            error={formik.touched.reference && Boolean(formik.errors.reference)}
+          />
+        </Grid>
+        <Grid item xs={12}>
+          <CustomTextField
+            name='name'
+            label={labels.name}
+            value={formik.values.name}
+            required
+            maxAccess={maxAccess}
+            readOnly={editMode}
+            onChange={formik.handleChange}
+            onClear={() => formik.setFieldValue('name', '')}
+            error={formik.touched.name && Boolean(formik.errors.name)}
+          />
+        </Grid>
+        <Grid item xs={12}>
+          <ResourceComboBox
+            endpointId={SystemRepository.Country.qry}
+            name='countryId'
+            label={labels.country}
+            valueField='recordId'
+            displayField='name'
+            readOnly={editMode}
+            displayFieldWidth={1}
+            columnsInDropDown={[
+              { key: 'reference', value: 'Reference' },
+              { key: 'name', value: 'Name' }
+            ]}
+            values={formik.values}
+            required
+            maxAccess={maxAccess}
+            onChange={(event, newValue) => {
+              formik.setFieldValue('stateId', null)
+              if (newValue) {
```

Comment:
why this if statement? I know it's present in code somewhere else by we can have  `formik.setFieldValue('countryId', newValue?.recordId || '')`